src/main/scala/com/databricks/spark/sql/perf/mllib/BenchmarkAlgorithm.scala

```diff
@@ -82,7 +90,9 @@ trait TestFromTraining {
     // Copy the context with a new seed.
     val ctx2 = ctx.params.randomSeed match {
       case Some(x) =>
-        val p = ctx.params.copy(randomSeed = Some(x + 1))
+        // Also set the number of examples to the number of test examples.
```

Comment:
This is odd; it makes me think passing a "context" to trainingDataSet() is not useful.  It would be simpler to pass just what is needed: the number of examples, SQLContext, etc.